addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/WebCacheDeceptionScanRule.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public void scan() {
+        if (initialTest()) {
+            HttpMessage authorisedMessage = getNewMsg();
+            try {
+                sendAndReceive(authorisedMessage);
+                String originalResponse =
+                        new String(authorisedMessage.getResponseBody().getBytes());
+                URI uri = authorisedMessage.getRequestHeader().getURI();
+                ArrayList<String> extensions = new ArrayList<String>();
+                String path = uri.getPath();
+                // checks whether the page with appended path gets cached or not
+                for (String ext : TEST_EXTENSIONS) {
+                    String newPath = path + "/test." + ext;
+                    uri.setPath(newPath);
+                    authorisedMessage.getRequestHeader().setURI(uri);
+                    sendAndReceive(authorisedMessage);
+                    String method = authorisedMessage.getRequestHeader().getMethod();
+                    String cachedResponse =
+                            new String(authorisedMessage.getResponseBody().getBytes());
+
+                    HttpMessage unauthorisedMessage = new HttpMessage(uri);
+                    unauthorisedMessage.getRequestHeader().setMethod(method);
+                    sendAndReceive(unauthorisedMessage);
+                    String unauthorisedCachedResponse =
+                            new String(unauthorisedMessage.getResponseBody().getBytes());
+                    // similarity between authorised and unauthorised response is checked
+                    if (checkSimilarity(unauthorisedCachedResponse, cachedResponse)) {
+                        extensions.add(ext);
+                    }
+                }
+                uri.setPath(path);
+                authorisedMessage.getRequestHeader().setURI(uri);
+                sendAndReceive(authorisedMessage);
+                if (!extensions.isEmpty()) {
+                    String attack = "";
+                    for (String ext : extensions) {
+                        attack = attack + "/test." + ext + ",";
+                    }
+                    newAlert()
+                            .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                            .setMessage(authorisedMessage)
+                            .setOtherInfo(
+                                    "Cached Authorised Response and Unauthorised Response are similar.")
+                            .setAttack(attack)
+                            .raise();
+                    return;
+                }
+            } catch (URIException e) {
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
This should use the logger (check other rules for examples). If the handling is going to be identical it may as well be a multi-catch.